src/tiled/tilesetdocument.cpp

```diff
@@ -347,6 +347,13 @@ void TilesetDocument::removeTiles(const QList<Tile *> &tiles)
     emit tilesetChanged(mTileset.data());
 }
 
+void TilesetDocument::relocateTile(const Tile *tile, int location)
+{
+    mTileset->relocateTile(tile->id(), location);
+    /* TODO: is that the right signal? */
```

Comment:
I think this is currently the right signal, though it isn't optimal or entirely correct. Ideally we'd send a signal before and after the change, such that the model can call `beginResetModel` and `endResetModel` in the appropriate moments. At the same time I'm trying to move away from all the separate signals, instead sending an specialized `ChangeEvent` in a unified `Document::changed` signal. I can help with this part of the change.

Reply:
I'll leave it as-is. If that's the best signal for this scenario (given how signal handling is currently implemented) I think it's better to make it consistent with the rest of the code. Signal handling improvements can come later in a separate change.